### PR TITLE
Implement feedback from 15366

### PR DESCRIPTION
**Pull request by deBhal**
Implements feedback from #15366

### CODE REVIEW

File: client/components/data/query-contact-details-cache/index.jsx [hunk 1]
```diff
@@ -16,9 +16,16 @@ import { requestContactDetailsCache } from 'state/domains/management/actions';
 
 class QueryContactDetailsCache extends Component {
 	componentWillMount() {
-		if ( this.props.isRequesting || ! isEmpty( this.props.contactDetailsCache ) ) {
+		const {
+			isRequesting,
+			contactDetailsCacheIsEmpty,
```
Comment (mcsf) [inline client/components/data/query-contact-details-cache/index.jsx]: propTypes declaration needs to be updated. :)

File: client/components/data/query-contact-details-cache/index.jsx [hunk 2]
```diff
@@ -16,9 +16,16 @@ import { requestContactDetailsCache } from 'state/domains/management/actions';
...
+		} = this.props;
+
+		if ( isRequesting || ! contactDetailsCacheIsEmpty ) {
 			return;
 		}
+
+		// Not destructured to disambiguate unbound action
```
Comment (mcsf) [inline client/components/data/query-contact-details-cache/index.jsx]: I feel like, if we start adding these types of comments, we'll end up with a big pile of unnecessary justifications. If someone actually sees this and cares so much that they'll try and "fix it", ESLint will yell at them, so it's fine. :) Let's remove the comment.

File: client/components/data/query-contact-details-cache/index.jsx (hunk 1 above)
Comment (deBhal) [inline client/components/data/query-contact-details-cache/index.jsx]: ✅ 

Turns out there's a rule for this. I've added [`react/prop-types`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md
) to my local `.eslintrc.js` to try it out and see if it's practical to add to our config.

**Comment by apeatling** [comment]
@debhal You got a thumbs up on this one, so let's merge.

**Comment by apeatling** [comment]
Reopen if/when ready to merge

**Pull request closed without merging**